packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/test/Providers/Definitions/ModelReaderWriterContextDefinitionTests.cs

```diff
@@ -188,6 +192,87 @@ public void ValidateModelReaderWriterBuildableAttributesIncludeDependencyModels(
             Assert.AreEqual(2, buildableAttributes.Count(), "Exactly two ModelReaderWriterBuildableAttributes should be generated for models with dependency references");
         }
 
+        [Test]
+        public void ValidateExperimentalModelAttributesAreTracked()
+        {
+            // Create an experimental model with ExperimentalAttribute
+            var experimentalModel = InputFactory.Model("ExperimentalModel", properties:
+            [
+                InputFactory.Property("Value", InputPrimitiveType.String)
+            ]);
+
+            var normalModel = InputFactory.Model("NormalModel", properties:
+            [
+                InputFactory.Property("Value", InputPrimitiveType.String)
+            ]);
+
+            var mockGenerator = MockHelpers.LoadMockGenerator(
+                inputModels: () => [experimentalModel, normalModel]);
+
+            // Add experimental attribute to the experimental model's provider
+            var experimentalModelProvider = mockGenerator.OutputLibrary.TypeProviders
+                .OfType<ModelProvider>()
+                .First(mp => mp.Name == "ExperimentalModel");
+
+            // Simulate the experimental attribute
+            var experimentalAttribute = new AttributeStatement(
+                new CSharpType(typeof(ExperimentalAttribute)), 
+                Literal("TYPESPEC001"));
+            
+            experimentalModelProvider.Update(attributes: [experimentalAttribute]);
+
+            var contextDefinition = new ModelReaderWriterContextDefinition();
+            var attributes = contextDefinition.Attributes;
+
+            Assert.IsNotNull(attributes);
+            Assert.AreEqual(2, attributes.Count, "Should have attributes for both models");
+
+            // Check that experimental info is tracked
+            Assert.AreEqual(1, contextDefinition._experimentalAttributeInfo.Count, 
+                "Should track one experimental attribute");
+            Assert.IsTrue(contextDefinition._experimentalAttributeInfo.ContainsValue("TYPESPEC001"),
+                "Should track the experimental key TYPESPEC001");
+        }
+
+        [Test]
+        public void ValidateExperimentalModelContextGeneratesWithPragmaWarnings()
```

Comment:
Lets add another test with multiple experimental models and multiple regular models.